src/helpers/api.ts

```diff
@@ -0,0 +1,161 @@
+    if (!exclude.find(pattern => pattern.test(config.url || ''))) {
+      console.groupCollapsed(`=> ${config.method?.toUpperCase()} ${config.url}`); // eslint-disable-line no-console
+      console.dir(clone(config)); // eslint-disable-line no-console
+      console.groupEnd(); // eslint-disable-line no-console
+    }
+
+    return config;
+  });
+
+  axios.interceptors.response.use((response: AxiosResponse) => {
+    if (!exclude.find(pattern => pattern.test(response.config.url || ''))) {
+      console.groupCollapsed(`<= ${response.config.method?.toUpperCase()} ${response.config.url}`); // eslint-disable-line no-console
+      console.dir(clone(response)); // eslint-disable-line no-console
+      console.groupEnd(); // eslint-disable-line no-console
+    }
+
+    return response;
+  });
+}
+
+/**
+ * Pushes an array of messages to the notification handler.
+ */
+function showNotifications(messages: IMessage[], options?: INotification): void {
+  if (!Array.isArray(messages)) {
+    return;
+  }
+
+  messages.forEach((message) => {
+    store.commit.notification.pushNotification({
+      ...options,
+      message,
+    });
+  });
+}
+
+/**
+ * Handles successful ajax requests.
+ *
+ * @returns {Object}
+ */
+function handleSuccess(response: AxiosResponse, options?: INotification): AxiosResponse {
+  const { messages } = response?.data || {};
+
+  if (messages) {
+    showNotifications(messages, options);
+  }
+
+  return response || {};
+}
+
+/**
+ * Handles axios error responses.
+ */
+function handleError(error: AxiosError, options?: INotification): Promise<AxiosError> {
+  const { messages } = error?.response?.data || {};
+
+  if (messages) {
+    showNotifications(messages, options);
+  } else {
+    store.dispatch.notification.showUnknownError();
+  }
+
+  return Promise.reject(error);
+}
+
+const api: IApi = {
+  /**
+   * Gets the url for the given 'urlKey'. The method also accepts an Object of interpolation values.
+   */
+  getUrl(urlKey: keyof typeof apiUrls, values: IUrlKeyValues): string {
+    let url = apiUrls[urlKey];
+
+    if (!url) {
+      throw new Error(`Unable to find an API url with the identifier ${urlKey}.`);
+    }
+
+    if (values) {
+      Object.entries(values).forEach(([key, value]) => {
+        url = url.replace(`{${key}}`, value);
+      });
+    }
+
+    return url;
+  },
+
+  /**
+   * Runs a get request with given url with given url params.
+   */
+  // eslint-disable-next-line max-len,@typescript-eslint/no-explicit-any
+  get(url: string, config: AxiosRequestConfig, notificationOptions: INotification): Promise<AxiosResponse<any> | AxiosError<any>> {
+    return axios
+      .get(url, config)
+      .then(response => handleSuccess(response, notificationOptions))
+      .catch(error => handleError(error, notificationOptions));
+  },
+
+  /**
+   * Runs a post request with a given url and payload.
+   */
+  // eslint-disable-next-line max-len, @typescript-eslint/no-explicit-any
```

Comment:
Please combine this eslint disabling with the one on the next line (I personally always use disabling on the same line as the related code, but since this is not possible with `ts-lint` I might have to change my behaviour here).